File: Backend/Models/SubscribedEmail.php
<?php

    class SubscribedEmail
    {
        private $database=null;

        public $Email;
        public $IP;
        public $SubscriptionDate;

        public function __construct($db)
        {
            $this->database=$db;
        }

        public function setSQLTurkeyTimeZone()
        {
            $query="SET time_zone = '+03:00' ;";
            mysqli_query($this->database, $query) or die("Time zone error...");
        }

        public function canThisUserMakeTodayASuscription()
        {
            $query="SELECT (SELECT COUNT(*) FROM SUBSCRIBED_EMAIL WHERE IP = '$this->IP' AND CAST(SubscriptionDate AS DATE) = DATE(NOW())  ) <=1";

            if ($result = mysqli_query($this->database, $query) or die("User subscription count query error...")) 
            {
                $row=mysqli_fetch_array($result, MYSQLI_NUM);
                return (int)$row[0];
            }
        }
		
		public function isValidEmail()
        {
            return (filter_var($this->Email, FILTER_VALIDATE_EMAIL)) ? 1 : 0;
        }

        public function create()
        {
			if($this->isValidEmail()==0)
			{
				echo "Invalid e-mail...\n";    
			}
		
			else
			{
				if($this->canThisUserMakeTodayASuscription()==1)
				{
					$this->setSQLTurkeyTimeZone();
				
					$query="INSERT INTO SUBSCRIBED_EMAIL VALUES('$this->Email', '$this->IP', '$this->SubscriptionDate')";
					$result=mysqli_query($this->database, $query);
                    if(!$result)
                    {
                        echo "Subscription insert error... " . mysqli_error($this->database);
                        die();
                    }
					
					echo "Subscription made! You will be notified of new posts to the e-mail you entered!";
				}
				
				else
				{
					echo "You passed the maximum subscriptions quota for today!\nEvery day a maximum of 2 suscriptions can be made for each device!";
				}	
			}
        }

        public function deleteSubscriptionWithThatEmail()
        {
            $query="DELETE FROM SUBSCRIBED_EMAIL WHERE Email = '$this->Email'";
            
            mysqli_query($this->database, $query) or die("Subscription delete error...");
			
			echo "Email ". $this->Email ." has been successfully unsubscribed!";
        }
		
		public function getAllUsersEmailsAsArray()
		{
			$usersEmailArray=array();
			
			$query="SELECT Email FROM SUBSCRIBED_EMAIL";
			
			if ($result = mysqli_query($this->database, $query) or die("Emails query error...")) 
            {
                while($row=mysqli_fetch_array($result, MYSQLI_ASSOC))
                {
                    $usersEmailArray[]=$row["Email"];
                }
            }

			return $usersEmailArray;
		}

		
		public function notifyAllUsersForNewPost($emailService, $newPostTitle, $newPostID, 
            $newPostImageURL, $newPostCategory, $newPostDate)
        {
			$newPostLink="https://andresrodriguez55.github.io/#/post/" . $newPostID . "/" . str_replace(" ","%20", $newPostTitle);

            $subject = "New Post of Andres's Blog!". $newPostTitle;
			
			$usersEmailArray=$this->getAllUsersEmailsAsArray();

			foreach($usersEmailArray as $userEmail)
			{
                $message = "
                <html>
                    <head>
                        <style>
                            .postGrid
                            {
                                background-color: rgba(230, 230, 230, 0.308);
                                position: relative;
                                -webkit-transition: all 400ms;
                            }

                            .postGridImage
                            {
                                display: block;
                                margin-left:auto;
                                margin-right:auto;
                                width: 50%;
                                filter: brightness(70%);
                            }

                            .postGridInformation
                            {
                                background-color: #000000;
                                color: white;
                                text-shadow: 2px 2px 2px #000000;
                                position: absolute;
                                bottom: 15%;
                                padding-left: 2%;
                                padding-right: 2%;
                                padding-top: 10px;
                                padding-bottom: 10px;
                                width: 48%;
                            }

                            .postGridTitle
                            {
                                font-size: 135%;
                                margin: 0;
                            }

                            .postGridDate
                            {
                                font-size: 82%;
                            }

                            .postGridCategory
                            {
                                font-size: 90%;
                            }
                        </style>
                    </head>
                    <body>
                        <div class='postGrid'> 
                            <img class='postGridImage' src='".$newPostImageURL."' />
                            <center>
                                <div class='postGridInformation'>
                                    <h2 class='postGridTitle'>".$newPostTitle."</h2>
                                    <p class='postGridCategory'>Category: ".$newPostCategory."</p>
                                    <p class='postGridDate'>".$newPostDate."</p>
                                </div>  
                            </center>
                        </div>
                        <a href='". $newPostLink ."'>Post link.</a>
                        <br>
                        <a href='https://".$_SERVER['HTTP_HOST']."/SubscribedEmail/DeleteEmail.php?Email=".$userEmail."'>
                            Unsubscribe
                        </a>          
                    </body>
                </html>
                ";

                $emailService->sendMail($message, $userEmail);
			}
        }
    }
?>

File: blogBackend/Models/SubscribedEmail.cs
using System;
using System.Collections.Generic;
using SendGrid;
using SendGrid.Helpers.Mail;


namespace blogBackend.Models
{
    public partial class SubscribedEmail
    {
        public string Email { get; set; } = null!;
        public DateTime SubscriptionDate { get; set; }
        
        private string buildNotificationEmailContent(Post post) 
        {
            string newPostUrl = Environment.GetEnvironmentVariable("PAGE_URL") + "post/" + post.Id.ToString() + "/" +
                post.Title.Replace(" ", "%20");

            string unsubscribeUrl = Environment.GetEnvironmentVariable("PAGE_URL") + "unsubscribe/" + this.Email;

            string htmlMessage = @"<html>
                    <head>
                        <style>
                            .postGrid
                            {
                                background-color: rgba(230, 230, 230, 0.308);
                                position: relative;
                                -webkit-transition: all 400ms;
                            }

                            .postGridImage
                            {
                                display: block;
                                margin-left:auto;
                                margin-right:auto;
                                width: 50%;
                                filter: brightness(70%);
                            }

                            .postGridInformation
                            {
                                background-color: #000000;
                                color: white;
                                text-shadow: 2px 2px 2px #000000;
                                position: absolute;
                                bottom: 15%;
                                padding-left: 2%;
                                padding-right: 2%;
                                padding-top: 10px;
                                padding-bottom: 10px;
                                width: 48%;
                            }

                            .postGridTitle
                            {
                                font-size: 135%;
                                margin: 0;
                            }

                            .postGridDate
                            {
                                font-size: 82%;
                            }

                            .postGridCategory
                            {
                                font-size: 90%;
                            }

                        </style>
                    </head>
                    <body>
                        <div class='postGrid'> 
                            <img class='postGridImage' src='" + post.CoverPhotoLink + @"' />
                            <center>
                                <div class='postGridInformation'>
                                    <h2 class='postGridTitle'>" + post.Title + @"</h2>
                                    <p class='postGridCategory'>Category: " + post.Category + @"</p>
                                    <p class='postGridDate'>" + post.PostDate + @"</p>
                                </div>  
                            </center>
                        </div>
                        <a href = '" + newPostUrl + @"' > Post link.</a>
                        <br><br>
                        <a href='" + unsubscribeUrl + @"' >
                            Unsubscribe 
                        </a>
                    </body>
                </html>";

            return htmlMessage;
        }

        public async Task<bool> notify(Post post)
        {
            var apiKey = Environment.GetEnvironmentVariable("SENDGRID_API_KEY");
            var client = new SendGridClient(apiKey);
            var hostEmail = Environment.GetEnvironmentVariable("EMAIL");
            var from = new EmailAddress(hostEmail, "Andres Personal Website");
            var to = new EmailAddress(this.Email, "User");
            var subject = "New Post of Andres's Blog!";
            var plainTextContent = "Click to see!";
            var htmlContent = buildNotificationEmailContent(post);
            var msg = MailHelper.CreateSingleEmail(from, to, subject, plainTextContent, htmlContent);

            var response = await client.SendEmailAsync(msg);
            Console.WriteLine(response);
            Console.WriteLine(response.StatusCode);
            Console.WriteLine(response.ToString());
            Console.WriteLine(response.Headers);
            return response.IsSuccessStatusCode;

        }
    }
}
